lockbox-ios/Presenter/ItemDetailPresenter.swift

```diff
@@ -151,9 +158,30 @@ class ItemDetailPresenter {
 
     private func setupNavigation() {
         self.view?.leftBarButtonTapped
-                .withLatestFrom(self.itemDetailStore.isEditing) { _, editing -> Action in
-                    return editing ? ItemDetailDisplayAction.viewMode : MainRouteAction.list
+                .withLatestFrom(self.itemDetailStore.isEditing) { _, editing -> Action? in
+                    if editing {
+                        self.view?.displayAlertController(
+                            buttons: [
+                                AlertActionButtonConfiguration(
+                                    title: Constant.string.cancel,
+                                    tapObserver: nil,
+                                    style: .cancel
+                                ),
+                                AlertActionButtonConfiguration(
+                                    title: Constant.string.discard,
+                                    tapObserver: self.discardChangesObserver,
+                                    style: .destructive)
+                            ],
+                            title: Constant.string.discardChangesTitle,
+                            message: Constant.string.discardChangesMessage,
+                            style: .alert,
+                            barButtonItem: nil)
+                    } else {
+                        return MainRouteAction.list
+                    }
+                    return nil
```

Comment:
why is there a `return nil` case? the `if editing` case will never fall through... is this a weird swift thing?

Reply:
Without it we see the error:

>Missing return in a closure expected to return 'Action?'